frontend/services/entities/hosts.ts

```diff
@@ -182,10 +216,18 @@ export default {
       path += `&policy_response=${policyResponse}`;
     }
 
-    if (!label && !policyId && softwareId) {
+    if (!label && !policyId && !mdmId && !mdmEnrollmentStatus && softwareId) {
       path += `&software_id=${softwareId}`;
     }
 
+    if (!label && !policyId && !softwareId && !mdmEnrollmentStatus && mdmId) {
+      path += `&mdm_id=${mdmId}`;
+    }
+
+    if (!label && !policyId && !softwareId && !mdmId && mdmEnrollmentStatus) {
+      path += `&mdm_enrollment_status=${mdmEnrollmentStatus}`;
+    }
+
```

Comment:
Same here re: conditional logic. Now that we're adding more and more filters we need to think about how to scale the number of filters in a maintainable way. 

cc @ghernandez345 FYI.

Reply:
@ghernandez345 built a cool build query params function we were going to pair up this morning and implement in a separate PR... but release work got in the way. Refactor coming soon.

Reply:
Cool, I like that approach! (I think that was based on something I wrote for another page 😊)